pkg/controller/operators/olm/operator.go

```diff
@@ -553,11 +625,77 @@ func (a *Operator) transitionCSVState(in v1alpha1.ClusterServiceVersion) (out *v
 			return
 		}
 	} else {
-		// TODO: This should never be the case.
+		// This should never be the case
 		logger.Info("no targets annotation defined for CSV")
 		out.SetPhaseWithEvent(v1alpha1.CSVPhaseFailed, v1alpha1.CSVReasonNoTargetNamespaces, err.Error(), now, a.recorder)
 	}
 
+	// Check for intersecting provided APIs in intersecting OperatorGroups
+	options := metav1.ListOptions{
+		FieldSelector: fmt.Sprintf("metadata.name!=%s,metadata.namespace!=%s", operatorGroup.GetName(), operatorGroup.GetNamespace()),
+	}
+	otherGroups, err := a.client.OperatorsV1alpha2().OperatorGroups(metav1.NamespaceAll).List(options)
+
+	groupSurface := resolver.NewOperatorGroup(*operatorGroup)
+	otherGroupSurfaces := resolver.NewOperatorGroupSurfaces(otherGroups.Items...)
+
+	operatorSurface, err := resolver.NewOperatorFromCSV(out)
+	if err != nil {
+		// TODO: Add failure status to CSV
+		syncError = err
+		return
+	}
+	providedAPIs := operatorSurface.ProvidedAPIs().StripPlural()
+
+	switch result := a.apiReconciler.Reconcile(providedAPIs, groupSurface, otherGroupSurfaces...); {
+	case operatorGroup.Spec.StaticProvidedAPIs && (result == resolver.AddAPIs || result == resolver.RemoveAPIs):
+		// Transition the CSV to FAILED with status reason "CannotModifyStaticOperatorGroupProvidedAPIs"
+		if out.Status.Reason != v1alpha1.CSVReasonInterOperatorGroupOwnerConflict {
+			logger.WithField("apis", providedAPIs).Warn("cannot modify provided apis of static provided api operatorgroup")
+			out.SetPhaseWithEvent(v1alpha1.CSVPhaseFailed, v1alpha1.CSVReasonCannotModifyStaticOperatorGroupProvidedAPIs, "static provided api operatorgroup cannot be modified by these apis", now, a.recorder)
+			a.cleanupCSVDeployments(logger, out)
```

Comment:
This stops any existing operator deployment from competing for resources in target namespaces.